SimpleWAWS/Code/BackgroundQueueManager.cs

```diff
@@ -158,43 +168,156 @@ private void HandleBackgroundOperation(BackgroundOperation operation)
                 default:
                     break;
             }
+        }
+
+        private void OnMaintainResourceGroupTimerElapsed(object state)
+        {
+            try
+            {
+                _jobHost.DoWork(async () =>
+                {
+                    await MaintainResourceGroupLists();
+                });
+            }
+            catch (Exception e)
+            {
+                SimpleTrace.Diagnostics.Fatal(e, "MaintainResourceGroupLists error, Count {Count}", Interlocked.Increment(ref _maintainResourceGroupListErrorCount));
+            }
 
         }
 
-        private void OnTimerElapsed(object state)
+        private void OnLogQueueStatsTimerElapsed(object state)
         {
             try
             {
                 _jobHost.DoWork(() =>
                 {
-                    MaintainResourceGroupLists();
-                    _logCounter++;
-                    if (_logCounter % 5 == 0)
+                    LogQueueStatistics();
+                });
+            }
+            catch (Exception e)
+            {
+                SimpleTrace.Diagnostics.Fatal(e, "LogQueueStatistics error, Count {Count}", Interlocked.Increment(ref _maintainResourceGroupListErrorCount));
+            }
+        }
+
+        private void OnCleanupSubscriptionsTimerElapsed(object state)
+        {
+            try
+            {
+                _jobHost.DoWork(async () =>
+                {
+                    if (!this.BackgroundInternalOperations.Any((a) => a.Value.Type == OperationType.SubscriptionLoad))
```

Comment:
If they are in the free queue they were okay. This method tries to delete the ones that couldn't be loaded. But thinking about it more. They may get loaded okay the first time, and then during the cleanuptimer tick, an http call fails, we may delete the resourcegroup and not remove from the loaded freequeue.